near-rs/integration-tests/src/tests.rs

```diff
@@ -35,222 +36,232 @@ async fn main() -> anyhow::Result<()> {
     Ok(())
 }
 
+pub struct ContractConnection<'a> {
+    user: &'a Account,
+    contract: &'a Contract,
+}
+
+impl <'a> ContractConnection<'_> {
+    async fn transact(&self, method: &str, json_args: serde_json::Value) -> Result<ExecutionFinalResult> {
+        self.user
+            .call(self.contract.id(), method)
+            .args_json(json_args)
+            .transact()
+            .await
+    }
+
+    async fn insert_grant(&self, json_args: serde_json::Value) -> Result<ExecutionFinalResult> {
+        self.transact("insert_grant", json_args).await
+    }
+
+    async fn delete_grant(&self, json_args: serde_json::Value) -> Result<ExecutionFinalResult> {
+        self.transact("delete_grant", json_args).await
+    }
+
+    async fn find_grants(&self, json_args: serde_json::Value) -> anyhow::Result<Vec<Grant>> {
+        Ok(
+            self.user
+                .call(self.contract.id(), "find_grants")
+                .args_json(json_args)
+                .view()
+                .await?
+                .json::<Vec<Grant>>().unwrap()
+        )
+    }
+}
+
+trait FallibleResult {
+    fn fails_with(&self, message: &str) -> bool;
+}
+
+impl FallibleResult for ExecutionFinalResult {
+    fn fails_with(&self, message: &str) -> bool {
+        self.is_failure() && self.clone().into_result().unwrap_err().to_string().contains(message)
+    }
```

Comment:
mini-nit: I'd sleep better at night if we didn't do a boolean assert without a help message. Maybe something like:
```suggestion
    fn assert_fails_with!(&self, expected: &str) -> bool {
        if !self.is_failure() { fail!("Call didn't fail. {self:?}") }

        let actual = self.clone().into_result().unwrap_err().to_string()

        assert!(
            actual.contains(expected),
            "Mismatched error string. Expected {expected:?} to appear in the error message {actual:?}.",
          )
    }
```